src/main/java/com/microsoft/aad/msal4j/OAuthAuthorizationGrant.java

```diff
@@ -14,42 +14,49 @@
 class OAuthAuthorizationGrant extends AbstractMsalAuthorizationGrant {
 
     private AuthorizationGrant grant;
-    private final Map<String, List<String>> params;
-
-    /**
-     * init standard scopes
-     */
-    private OAuthAuthorizationGrant() {
-        params = new LinkedHashMap<>();
-
-        params.put(SCOPE_PARAM_NAME, Collections.singletonList(COMMON_SCOPES_PARAM));
-    }
+    private final Map<String, List<String>> params = new LinkedHashMap<>();
 
     OAuthAuthorizationGrant(final AuthorizationGrant grant, Set<String> scopesSet, ClaimsRequest claims) {
         this(grant, scopesSet != null ? String.join(" ", scopesSet) : null, claims);
     }
 
-    OAuthAuthorizationGrant(final AuthorizationGrant grant, String scopes, ClaimsRequest claims) {
-        this();
-        this.grant = grant;
-
+    String addCommonScopes(String scopes){
+        String allScopes = COMMON_SCOPES_PARAM;
 
         if (!StringHelper.isBlank(scopes)) {
-            this.scopes = scopes;
-            params.put(SCOPE_PARAM_NAME,
-                    Collections.singletonList(String.join(" ",params.get(SCOPE_PARAM_NAME)) + SCOPES_DELIMITER + scopes));
+            allScopes = allScopes + SCOPES_DELIMITER + scopes;
         }
+        return allScopes;
+    }
+
+    OAuthAuthorizationGrant(final AuthorizationGrant grant, String scopes, ClaimsRequest claims) {
+        this.grant = grant;
+
+        String allScopes = addCommonScopes(scopes);
+        this.scopes = allScopes;
+        params.put(SCOPE_PARAM_NAME, Collections.singletonList(allScopes));
 
         if (claims != null) {
             this.claims = claims;
             params.put("claims", Collections.singletonList(claims.formatAsJSONString()));
         }
     }
 
-    OAuthAuthorizationGrant(final AuthorizationGrant grant,
-                                final Map<String, List<String>> params) {
-        this();
+    OAuthAuthorizationGrant(AuthorizationGrant grant, String scopes, Map<String, List<String>> extraParams) {
         this.grant = grant;
+
+        String allScopes = addCommonScopes(scopes);
+        this.scopes = allScopes;
```

Comment:
```suggestion
        this.scopes = addCommonScopes(scopes);
```

Reply:
Requires an update on line 50 as well.

Reply:
what do you mean ?

Reply:
Just meant that instead of setting `allScopes` here, you can get rid of it and just use `this.scopes`.